frame/contracts/src/tests.rs

```diff
@@ -143,12 +150,12 @@ impl Default for TestExtension {
 }
 
 impl ChainExtension<Test> for TestExtension {
-	fn call<E>(func_id: u32, env: Environment<E, InitState>) -> ExtensionResult<RetVal>
+	fn call<E>(&mut self, func_id: u32, env: Environment<E, InitState>) -> ExtensionResult<RetVal>
 	where
 		E: Ext<T = Test>,
 		<E::T as SysConfig>::AccountId: UncheckedFrom<<E::T as SysConfig>::Hash> + AsRef<[u8]>,
 	{
-		match func_id {
+		match func_id & 0x00FF {
```

Comment:
Maybe we can have something like: 

```rust
pub const FUNC_ID_MASK = 0x00FF;
```

Reply:
Done. That seemed a little low level. I replaced the parameter by two functions on `env`.